doc/guides/west/manifest.rst

```diff
@@ -0,0 +1,281 @@
+  together:
+
+  - ``P2`` stays the same from ``A → B``, as do ``P1`` and ``P3`` from ``F →
+    G``.
+  - ``P3`` moves forward from ``A → B``.
+  - ``P3`` moves backward from ``C → D``.
+
+  One use for moving backward in project history is to "revert" a regression by
+  going back to a revision before it was introduced.
+
+- Project commits can be "skipped": ``P3`` moves forward multiple commits in
+  its history from ``B → C``.
+
+- In the above diagram, no project has two revisions "at the same time": every
+  manifest file refers to exactly one commit in the projects it cares
+  about. This can be relaxed by using a branch name as a manifest revision, at
+  the cost of being able to bisect manifest repository history.
+
+Manifest Files
+**************
+
+A west manifest is a YAML file named :file:`west.yml`. Manifests have two
+top-level "sections", ``west`` and ``manifest``, like this:
+
+.. code-block:: yaml
+
+   west:
+     # contents of west section
+   manifest:
+     # contents of manifest section
+
+In YAML terms, the manifest file contains a mapping, with two keys relevant to
+west at top level. These keys are the scalar strings ``west`` and
+``manifest``. Their contents are described next.
+
+West Section
+============
+
+.. note::
+
+   Support for this feature will be removed in a future version of west, when
+   the west repository is no longer cloned into the installation.
+
+The ``west`` section specifies the URL and revision of the west repository
+which is cloned into the installation. For example:
+
+.. code-block:: yaml
+
+   west:
+     url: https://example.com/west
+     revision: v0.5.6
+
+This specifies cloning the west repository from URL
+``https://example.com/west`` (any URL accepted by ``git clone`` will work), at
+revision ``v0.5.6``. The revision can be a Git branch, tag, or SHA.
+
+That is, the west section also contains a mapping, with permitted keys ``url``
+and ``revision``. These specify the fetch URL and Git revision for the west
+repository to clone into the installation, as described in
+:ref:`west-struct`. If not given, the default URL is
+https://github.com/zephyrproject-rtos/west, and the default revision is
+``master``.
+
+The file :file:`west-schema.yml` in the west source code repository contains a
+pykwalify schema for this section's contents.
+
+Manifest Section
+================
+
+This is the main section in the manifest file. There are four subsections:
+``defaults``, ``remotes``, ``projects``, and ``self``. In YAML terms, the value
+of the ``manifest`` key is also a mapping, with these "subsections" as keys.
+For example:
+
+.. code-block:: yaml
+
+   manifest:
+     defaults:
+       # contents of defaults subsection
+     remotes:
+       # contents of remotes subsection
+     projects:
+       # contents of projects subsection
+     self:
+       # contents of self subsection
+
+The ``remotes`` and ``projects`` subsections are the only mandatory ones, so
+we'll cover them first.
```

Comment:
Would it be possible to have `modules:` and `projects:` as alternative names for the same section?

Reply:
Not without changing the way west works now. The purpose of this PR is just to document west.  Please open a west issue if you'd like to discuss manifest schema changes.